experimental/devices/mfrc522/commands/low_level.go

```diff
@@ -0,0 +1,293 @@
+		return nil, err
+	}
+
+	dev := &LowLevel{
+		spiDev:   spiDev,
+		irqPin:   irqPin,
+		resetPin: resetPin,
+	}
+
+	return dev, nil
+}
+
+// LowLevel is a low-level handler of a MFRC522 RFID reader.
+type LowLevel struct {
+	resetPin gpio.PinOut
+	irqPin   gpio.PinIn
+	spiDev   spi.Conn
+}
+
+// String implements conn.Resource.
+func (r *LowLevel) String() string {
+	return fmt.Sprintf("Mifare MFRC522 [bus: %v, reset pin: %s, irq pin: %s]",
+		r.spiDev, r.resetPin.Name(), r.irqPin.Name())
+}
+
+// DevWrite sends data to a device.
+func (r *LowLevel) DevWrite(address int, data byte) error {
+	newData := []byte{(byte(address) << 1) & 0x7E, data}
+	return r.spiDev.Tx(newData, nil)
+}
+
+// DevRead gets data from a device.
+func (r *LowLevel) DevRead(address int) (byte, error) {
+	data := []byte{((byte(address) << 1) & 0x7E) | 0x80, 0}
+	out := make([]byte, len(data))
+	if err := r.spiDev.Tx(data, out); err != nil {
+		return 0, err
+	}
+	return out[1], nil
+}
+
+// CRC calculates the CRC of the data using the card chip.
+func (r *LowLevel) CRC(inData []byte) ([]byte, error) {
+	if err := r.ClearBitmask(DivIrqReg, 0x04); err != nil {
+		return nil, err
+	}
+	if err := r.SetBitmask(FIFOLevelReg, 0x80); err != nil {
+		return nil, err
+	}
+	for _, v := range inData {
+		if err := r.DevWrite(FIFODataReg, v); err != nil {
+			return nil, err
+		}
+	}
+	if err := r.DevWrite(CommandReg, PCD_CALCCRC); err != nil {
+		return nil, err
+	}
+	for i := byte(0xFF); i > 0; i-- {
+		n, err := r.DevRead(DivIrqReg)
+		if err != nil {
+			return nil, err
+		}
+		if n&0x04 > 0 {
+			break
+		}
+	}
+	lsb, err := r.DevRead(CRCResultRegL)
+	if err != nil {
+		return nil, err
+	}
+
+	msb, err := r.DevRead(CRCResultRegM)
+	if err != nil {
+		return nil, err
+	}
+	return []byte{lsb, msb}, nil
+}
+
+// SetBitmask sets register bit.
+func (r *LowLevel) SetBitmask(address, mask int) error {
+	current, err := r.DevRead(address)
+	if err != nil {
+		return err
+	}
+	return r.DevWrite(address, current|byte(mask))
+}
+
+// ClearBitmask clears register bit.
+func (r *LowLevel) ClearBitmask(address, mask int) error {
+	current, err := r.DevRead(address)
+	if err != nil {
+		return err
+	}
+	return r.DevWrite(address, current&^byte(mask))
+}
+
+// StopCrypto stops the crypto chip.
+func (r *LowLevel) StopCrypto() error {
+	return r.ClearBitmask(Status2Reg, 0x08)
+}
+
+// WaitForEdge waits for an IRQ pin to strobe.
+func (r *LowLevel) WaitForEdge(timeout time.Duration) bool {
+	return r.irqPin.WaitForEdge(timeout)
+}
+
+// Auth authenticate the card fof the sector/block using the provided data.
+//
+// 	mode - the authentication mode.
+// 	sector - the sector to authenticate on.
+// 	block - the block within sector to authenticate.
+// 	sectorKey - the key to be used for accessing the sector data.
+// 	serial - the serial of the card.
+func (r *LowLevel) Auth(mode byte, blockAddress byte, sectorKey [6]byte, serial []byte) (AuthStatus, error) {
+	buffer := make([]byte, 2)
```

Comment:
Here's a way to not allocate memory on the heap, but that's really a micro optimization so you can ignore. :)
```
var b [32]byte{mode, blockAddress}
buf := append(b[:2], sectorKey[:]...)
buf = append(buf, serial...)
```